server/src/main/java/org/elasticsearch/search/aggregations/metrics/TDigestState.java

```diff
@@ -7,19 +7,18 @@
  */
 package org.elasticsearch.search.aggregations.metrics;
 
-import com.tdunning.math.stats.AVLTreeDigest;
-import com.tdunning.math.stats.Centroid;
-
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.tdigest.Centroid;
+import org.elasticsearch.tdigest.MergingDigest;
 
 import java.io.IOException;
 import java.util.Iterator;
 
 /**
- * Extension of {@link com.tdunning.math.stats.TDigest} with custom serialization.
+ * Extension of {@link org.elasticsearch.tdigest.TDigest} with custom serialization.
  */
-public class TDigestState extends AVLTreeDigest {
+public class TDigestState extends MergingDigest {
```

Comment:
So the wire format between avl and merging can stay the same?

Reply:
Right, serialization is handled by this class so there are no compatibility issues - other than the changes in quantile calculations.